### PR TITLE
Quirks are passed an incoming client when applied, allowing quirks to read preferences in `add` and `add_unique`. Renders visual quirks on the preference menu dummy.

### LINKED ISSUE #72141: Quirk preferences do not work roundstart
The "Nearsighted" quirk has a preference option to change between differing styles. Joining the round while it has already started will give you the correct kind of pair of glasses, while readying up and joining roundstart will grant you the default prescription glasses.

Latejoin:
dreamseeker_qkqhrqrIUx

Roundstart:
dreamseeker_2oT67hcpkI

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Closes #72141

Roundstart humans weren't in control of their mob by the time `AssignQuirks` was called. 
The call chain for roundstart goes `create_characters` -> `equip_characters` -> `AssignQuirks` -> `transfer_characters`.
For latejoin, `create_character` -> `transfer_character` -> `AssignQuirks`.

I could simply move around the call chain, but that feels much more fragile and more liable to cause other issues to me. So instead, I simply allowed add quirk to be passed a client, so that a client's quirks can be applied to a mob they are not currently inhabiting. 

In doing this, it became possible to show visual quirks on the prefs dummy, like nearsighted glasses and heterochromia. So I put in a little work to accomplish that as well. 

image

Along side, some refactoring and documentation for quirk datums. 

## Why It's Good For The Game

People get what they expect on roundstart. 

## Changelog

:cl: Melbert
qol: The preview dummy in the preferences menu now shows some visual quirks, like Heterochromia or nearsighted. 
fix: Fixed some quirks which read a preference roundstart not applying the preference.
refactor: Refactored some bits of quirk datums and the quirk application process. 
/:cl:

### CODE REVIEW

**Comment by iwishforducks** [comment]
Needs to be updated with https://github.com/tgstation/tgstation/pull/72143 being recently merged, I believe.

File: code/datums/quirks/neutral_quirks.dm [hunk 1]
```diff
@@ -192,36 +192,27 @@
...
-	link_to_holder()
-
-/datum/quirk/heterochromatic/remove()
-	UnregisterSignal(quirk_holder, COMSIG_CARBON_LOSE_ORGAN)
-
-/datum/quirk/heterochromatic/proc/link_to_holder()
+/// Applies the passed color to this mob's eyes
+/datum/quirk/heterochromatic/proc/apply_heterochromatic_eyes(color)
 	var/mob/living/carbon/human/human_holder = quirk_holder
+	var/was_not_hetero = !human_holder.eye_color_heterochromatic
```
Comment (Time-Green) [inline code/datums/quirks/neutral_quirks.dm:213]: hmmm

**Review by Time-Green — APPROVED** [review]

**Comment by Mothblocks** [comment]
are we sure this did not need to be updated with the new quirk like ducks said? @Time-Green @MrMelbert

**Comment by Time-Green** [comment]
> are we sure this did not need to be updated with the new quirk like ducks said? @Time-Green @MrMelbert

My sleep deprived brain might’ve read it wrong. I don’t think it shows up on the pref screen then?

**Comment by Mothblocks** [comment]
That and also probably doesn't work roundstart

**Pull request merged**